src/components/projects/vault/VaultItemEdit.js

```diff
@@ -0,0 +1,127 @@
+import { Input, Table, Tbody, Td, Th, Thead, Tr, Select, Checkbox, Button } from '@chakra-ui/react';
+import { actionCompletedToast, errorToast } from "components/ui/toast";
+import { useState } from "react";
+import secureApiFetch from "services/api";
+import { useNavigate, useParams } from "react-router-dom";
+import Vault from 'models/Vault';
+
+const VaultItemEdit = () => {
+    const { projectId, vaultItemId} = useParams();
+    const navigate = useNavigate();
+
+    const [item, setVaultItem] = useState(new Vault());
+    const [password, setPassword] = useState(null);
+
+    const onVaultItemFormChange = ev => {
+        const value = ev.target.type === 'checkbox' ? ev.target.checked : ev.target.value;
+        setVaultItem({ ...item, [ev.target.name]: value });
+    }
+
+    const onFormSubmit = ev => {
+        ev.preventDefault();
+
+        item.password = password;
+
+        secureApiFetch(`/vault/${projectId}/${vaultItemId}`, { method: 'PUT', body: JSON.stringify(item) })
+            .then(resp => {
+                if (resp.status === 201) {
+                    setVaultItem(new Vault());
+                    setPassword(null);
+                    actionCompletedToast(`The vault item has been modified.`);
+                    navigate(`/projects/${projectId}`);
+                } else {
+                    errorToast("The vault item could not be saved. Review the form data or check the application logs.")
+                }
+            })
+    }
+
+    const onPasswordProvided = ev => {
+        ev.preventDefault();
+
+        secureApiFetch(`/vault/${projectId}/${vaultItemId}`, { method: 'POST', body: JSON.stringify({ 'password' : password }) })
+            .then(response => response.json())
+            .then(json => {
+                debugger;
+                if (json['success'] === false)
+                {
+                    errorToast("Seems like a wrong password.");
+                    setPassword(null);
+                }
+                else 
+                {
+                    var newItem = new Vault();
+                    newItem.name = json['name'];
+                    newItem.note = json['note'];
+                    newItem.value = json['value'];
+                    newItem.type = json['type'];
+                    newItem.reportable = json['reportable'];
```

Comment:
You could type less characters in this way:
```js
                    newItem.name = json.name;
                    newItem.note = json.note;
                    ....
```

Reply:
nice, I did not know this. Thank you!